src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProvider.php

```diff
@@ -197,14 +175,14 @@ public function getParentCategory($categoryName)
     /**
      * Re-organize category data into a Menu item.
      *
-     * @param $menu
-     * @param $name
+     * @param string $menu
+     * @param string $name
      * @param array $moduleIds
-     * @param null $tab
+     * @param string $tab
      *
-     * @return object
+     * @return stdClass
      */
-    private function createMenuObject($menu, $name, $moduleIds = [], $tab = null)
+    private function createMenuObject(string $menu, string $name, array $moduleIds = [], ?string $tab = null): stdClass
     {
         return (object) array(
```

Comment:
Little bit overkill because it create and need an stdClass :/

Reply:
But I'll add unit tests

Reply:
You're right it's probably overkill, it just feel weird to use and object, if we don't use strong typed object I guess just using an array would do the job no? This would avoid weird syntaxes like `$categories['categories']->subMenu[self::CATEGORY_THEME]->modules` and the class could use `$categories = json_decode(json_encode($categories));` right before the return (since it's already present in the class)

Reply:
This was the original behaviour, and I can make a json_deciode / json_encode because we store `Module` objects into the `modules` key :sweat_smile: 
Playing with collection would have been much better :(

Reply:
Yeah that's why I suggested creating a real class, but I guess this will be an improvement for later